knowledge/guidelines.rst

```diff
@@ -0,0 +1,50 @@
+Core guidelines
+===============
+
+
+Good practices
+--------------
+
+- The recipes' ``generate()`` method purpose is to prepare the build as much as possible.
+  Users calling ``conan install`` will execute this method, and the generated files should
+  allow users to do "native" builds (calling directly "cmake", "meson", etc.) as easy as possible.
+  Thus, avoiding as much as possible any logic in the ``build()`` method, and moving it to
+  the ``generate()`` method helps developers achieve the same build locally as the one that 
+  would be produced by a ``conan create`` build in the local cache.
+- Always use your own profiles in production, instead of relying on the auto-detected profile,
+  as the output of such auto detection can vary over time, resulting in unexpected results.
+  Profiles (and many other configuration), can be managed with ``conan config install``.
+- Developers should not be able to upload to "development" and "production" repositories in the server.
+  Only CI builds have write permissions in the server. Developers should only have read permissions and 
+  at most to some "playground" repositories used to work and share things with colleagues,
+  but which packages are never used, moved or copied to the development or production repositories.
+- The ``test_package`` purpose is to check that the package has been correctly created (that is, 
```

Comment:
It would be nice to add a link here as well: https://docs.conan.io/en/2.0/tutorial/creating_packages/test_conan_packages.html